resolver/resolver.cc

```diff
@@ -2799,7 +2785,8 @@ class ResolveTypeMembersAndFieldsWalk {
                         todoResolveCastItems_.emplace_back(move(item));
                     }
 
-                    return ast::MK::InsSeq1(send.loc, move(typeExpr), move(cast));
+                    tree = ast::MK::InsSeq1(send.loc, move(typeExpr), move(cast));
+                    return;
```

Comment:
This drops the old one before overwriting it right?

This API doesn't even seem that annoying. I was expecting to not like it that much, but I think not having to write `return tree` everywhere to early exit is actually totally good compensation.

Reply:
That's correct; this line just going to invoke `ExpressionPtr::operator=(ExpressionPtr &&)`, which handles the memory management.